### PR TITLE
Add the DBus method IsNetworkRequired

### PR DESCRIPTION
**Pull request by poncovka**
Call the DBus method IsNetworkRequired of a payload module to check whether
the payload's sources require a network.

### CODE REVIEW

File: pyanaconda/modules/payloads/source/utils.py [hunk 1]
```diff
@@ -27,7 +27,20 @@
...
+def has_network_protocol(url):
+    """Does the given URL have a network protocol?
+
+    :param url: a string with URL
+    :return: True or False
+    """
+    if not url:
+        return False
+
+    network_protocols = ["http:", "ftp:", "nfs:", "nfsiso:"]
```
Comment (jstodola) [inline pyanaconda/modules/payloads/source/utils.py:42]: https is missing

Comment (VladimirSlavik) [reply pyanaconda/modules/payloads/source/utils.py:42]: See #2321...

File: pyanaconda/modules/payloads/source/source_base.py [hunk 2]
```diff
@@ -58,6 +58,14 @@ def description(self):
         """
         pass
 
+    @property
+    def network_required(self):
+        """Does the source require a network?
+
+        :return: True or False
+        """
+        return False
```
Comment (VladimirSlavik) [inline pyanaconda/modules/payloads/source/source_base.py:67]: Maybe this should be abstract like the rest of these properties? Not sure.

File: pyanaconda/modules/payloads/source/repo_files/repo_files.py [hunk 3]
```diff
@@ -43,6 +43,14 @@ def description(self):
         """Get description of this source."""
         return _("Closest mirror")
 
+    @property
+    def network_required(self):
+        """Does the source require a network?
+
+        :return: True or False
+        """
+        return True
```
Comment (VladimirSlavik) [inline pyanaconda/modules/payloads/source/repo_files/repo_files.py:52]: Is this true in the mathematical sense, or just the best approximation? I could see repo files with only file://, but maybe these are special cases handled by other sources?

File: pyanaconda/modules/payloads/source/utils.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/modules/payloads/source/utils.py]: Fixed. I have used the commit from https://github.com/rhinstaller/anaconda/pull/2321.

File: pyanaconda/modules/payloads/source/source_base.py (hunk 2 above)
Comment (poncovka) [reply pyanaconda/modules/payloads/source/source_base.py]: Fixed.

File: pyanaconda/modules/payloads/source/repo_files/repo_files.py (hunk 3 above)
Comment (poncovka) [reply pyanaconda/modules/payloads/source/repo_files/repo_files.py:52]: It is the best approximation. This code will replace the implementation of the property `needs_network` from `DNFPayload` and we assume there that closest mirror requires network. See: https://github.com/rhinstaller/anaconda/blob/master/pyanaconda/payload/dnf/payload.py#L158

**Comment by poncovka** [comment]
Rebased and fixed.

File: pyanaconda/modules/payloads/source/utils.py (hunk 1 above)
Comment (VladimirSlavik) [reply pyanaconda/modules/payloads/source/utils.py]: Thank you!

File: pyanaconda/modules/payloads/source/repo_files/repo_files.py (hunk 3 above)
Comment (VladimirSlavik) [reply pyanaconda/modules/payloads/source/repo_files/repo_files.py:52]: Sorry, here I'm stupid again - of course that's how the code worked previously. Thank you for the explanation!

**Pull request merged**